core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -1111,6 +1111,10 @@ class ZKCheckedEphemeral(path: String,
                             case Code.INVALIDACL =>
                               error("Invalid ACL")
                               setResult(Code.INVALIDACL)
+                            case Code.NOAUTH =>
+                              // Allows old consumer to register with secure ZK
+                              error("NOAUTH raised on path %s (AUTH may not be needed to create children)".format(path))
+                              createRecursive(path, suffix)
```

Comment:
Not that it is correct but its worth noting that this is the current behavior as well. Each node in the path is evaluated regardless if a NOAUTH(or any other non-OK) return code is encountered or not. The end user will not be privy to this without delving into the logs either. 

What makes it confusing is the fact that the return code was returned for a node which was not the intended target. The exception that ultimately surfaces is NOATUH on `/consumer`. While its true that the consumer does not have adequate privileges on `/` this was not the intent of the command. This gives off the impression the client does not have permissions to create nodes under `/consumer` as this is ultimately the target. 

I'd argue that evaluating the return code for any node that is not the leaf node makes this difficult to troubleshoot. For example, seeing that NOAUTH is raised on `/consumer` gives off the impression you do not have permission to create nodes under `/consumer`. This is not actually the case, the user has adequate permissions to create nodes under `/consumer` just not under `/`. Whats more `/consumer` isn't the target so it's less than intuitive to think that the path was being evaluated as if we were creating that node. This is still only visible from the logs as well. 

At any rate if you think its better to check for NodeExists inside the NOAUTH as well I can do that. I'm not sure I agree this makes it more friendly to troubleshoot but I'll concede if you think it's best.